### ISSUE TITLE
📝 Incorrect error line number on vue files

### ISSUE THREAD

**Issue by davidglezz** [biomejs/biome#4217]
### Environment information

```bash
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.17.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.2"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0


```

Cli error on unrelated line
![image](https://github.com/user-attachments/assets/d1683d04-f5bf-4c51-abbf-7d0e665ca342)

Related?: https://github.com/biomejs/biome/pull/3494

### Configuration

```JSON
no config / defaults
```

### Playground link

https://biomejs.dev/playground/?enabledLinting=false&importSortingEnabled=false&files.example.vue=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AHMAZQBjAHQAaQBvAG4APgAKACAAIAAgACAAIAAgACAAIAAgACAAPABoADEAPgBOAG8AdAAgAGYAbwByAG0AYQB0AHQAZQBkACAAOgAoADwALwBoADEAPgAKACAAIAA8AC8AcwBlAGMAdABpAG8AbgA%2BAAoAPAAvAHQAZQBtAHAAbABhAHQAZQA%2BAAoACgA8AHMAYwByAGkAcAB0ACAAbABhAG4AZwA9ACIAdABzACIAIABzAGUAdAB1AHAAPgAKAGMAbwBuAHMAdAAgAGQAYQB0AGEAQwBvAG4AZgBpAGcAIAA9ACAAWwAKACAAIAAvACoAKgAKACAAIAAgACoAIAAqAAoAIAAgACAAKgAgACoAIAAqAAoAIAAgACAAKgAgACoAIAAqACAAKgAKACAAIAAgACoAIAAqACAAKgAgACoAIAAqAAoAIAAgACAAKgAgACoAIAAqACAAKgAKACAAIAAgACoAIAAqACAAKgAKACAAIAAgACoAIAAqAAoAIAAgACAAKgAvAAoAXQA7AAoACgB0AHIAeQAgAHsACgAgACAALwAqACAAdAByAHkAIAAqAC8ACgB9ACAAYwBhAHQAYwBoACAAKABlAHIAcgA6ACAARQByAHIAbwByACAAfAAgAGEAbgB5ACkAIAB7AAoAIAAgAC8AKgAgAC8AXAAgAEMAYQB0AGMAaAAgAGMAbABhAHUAcwBlACAAdgBhAHIAaQBhAGIAbABlACAAdAB5AHAAZQAgAGEAbgBuAG8AdABhAHQAaQBvAG4AIABtAHUAcwB0ACAAYgBlACAAJwBhAG4AeQAnACAAbwByACAAJwB1AG4AawBuAG8AdwBuACcAIABpAGYAIABzAHAAZQBjAGkAZgBpAGUAZAAuACAAKgAvAAoAfQAKADwALwBzAGMAcgBpAHAAdAA%2BAAoACgA8AHMAdAB5AGwAZQAgAGwAYQBuAGcAPQAiAHMAYwBzAHMAIgAgAHMAYwBvAHAAZQBkAD4ACgAgACAALwAqACAAbgBvAHQAIABmAG8AcgBtAGEAdAB0AGUAZAAgADoAKAAgACoALwAKACAAIAAuAHQAZQBzAHQAIAAgACAAIAAgACAAewAgAGMAbwBsAG8AcgA6ACAAIAAgACAAcgBlAGQAOwAgAH0ACgA8AC8AcwB0AHkAbABlAD4ACgA%3D

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
